plotly/tools.py

```diff
@@ -1464,31 +2285,39 @@ def _find_intermediate_color(lowcolor, highcolor, intermed):
         diff_1 = float(highcolor[1] - lowcolor[1])
         diff_2 = float(highcolor[2] - lowcolor[2])
 
-        new_tuple = (lowcolor[0] + intermed*diff_0,
-                     lowcolor[1] + intermed*diff_1,
-                     lowcolor[2] + intermed*diff_2)
-
-        return new_tuple
+        inter_colors = (lowcolor[0] + intermed * diff_0,
+                        lowcolor[1] + intermed * diff_1,
+                        lowcolor[2] + intermed * diff_2)
+        return inter_colors
 
     @staticmethod
     def _unconvert_from_RGB_255(colors):
         """
         Return a tuple where each element gets divided by 255
 
-        Takes a list of color tuples where each element is between 0 and 255
-        and returns the same list where each tuple element is normalized to be
-        between 0 and 1
+        Takes a (list of) color tuple(s) where each element is between 0 and
+        255. Returns the same tuples where each tuple element is normalized to
+        a value between 0 and 1
 
         """
-        un_rgb_colors = []
-        for color in colors:
-            un_rgb_color = (color[0]/(255.0),
-                            color[1]/(255.0),
-                            color[2]/(255.0))
+        if isinstance(colors, tuple):
```

Comment:
🐄 This is pretty un-duck-type-y. What's preventing a future tuple of tuples here?